packages/server-wallet/src/wallet/response-builder.ts

```diff
@@ -0,0 +1,204 @@
+import {ChannelResult} from '@statechannels/client-api-schema';
+import {Participant, serializeMessage, SignedState} from '@statechannels/wallet-core';
+
+import {Channel} from '../models/channel';
+import {DBObjective, toWireObjective} from '../models/objective';
+import {Outgoing} from '../protocols/actions';
+import {mergeChannelResults, mergeOutgoing} from '../utilities/messaging';
+import {WALLET_VERSION} from '../version';
+
+import {MultipleChannelOutput, SingleChannelOutput, WalletEvent} from '.';
+
+export interface ResponseBuilder {
+  /**
+   * Queues channel for notification to user
+   */
+  channelUpdated: (channel: Channel) => void;
+
+  /**
+   * Same as channelUpdated but accepts a channelResult instead of a Channel
+   *
+   * Plan to deprecate.
+   */
+  channelUpdatedResult: (channelResult: ChannelResult) => void;
+
+  /**
+   * Queues state for sending to opponent
+   */
+  stateSigned: (state: SignedState, myIndex: number, channelId?: string) => void;
+
+  /**
+   * Queues objectives for sending to opponent
+   */
+  objectiveCreated: (objective: DBObjective, myIndex: number, participants: Participant[]) => void;
+
+  /**
+   * Queues objectives for approval by the user
+   */
+  objectiveReceived: (objective: DBObjective) => void;
```

Comment:
I'm wondering if there are more intuitive names for these methods. For example `queueChannel`, `queueSignedState` etc.  You could even use function overloading and have a single `queue` method that you can pass various types into.

Reply:
The original intent was that the style would be as declarative as possible: the wallet would tell the response that something has happened, and the response would then decide what to do. That was the thinking behind having all the methods named as "X happened". I'm open to a change though. It sounds like the full set would become:

* `queueChannel`
* `queueChannelResult`
* `queueState`
* `queueCreatedObjective`
* `queueReceivedObjective`
* `queueCompletedObjective`
* `queueChannelRequest`

Do these sound reasonable?

Reply:
Yes that sounds reasonable. I don't feel super strongly about it, as the original intent you mentioned also seems reasonable -- and would probably make it a bit easier if we want to change the response format in the future. 

There is, however, a bit of a disconnect between that and the docstrings you put in the interface -- would be good to keep those in tighter sync.

Reply:
Made the change. I think I prefer it. d2583cf

Reply:
+1 for docstrings, by the way. I think we should be doing more of that.